src/fastalloc/mod.rs

```diff
@@ -406,6 +491,12 @@ impl<'a, F: Function> Env<'a, F> {
             && self.edits.scratch_regs[class].is_none()
         {
             self.alloc_scratch_reg(inst, class, pos)?;
+            self.num_available_pregs[ExclusiveOperandPos::Both][class] =
+                0i16.max(self.num_available_pregs[ExclusiveOperandPos::Both][class] - 1);
```

Comment:
We can use `saturating_sub()` here and below, I think, to avoid the underflow on `-`.

Reply:
They are `i16`s and`saturating_sub` is relative to the type's minimum value, but I've clarified that code anyways.

Reply:
Ah, that's right; thanks for the small refactor there!